File: view/src/test/java/by/rekuts/travelagency/view/TestSecurityConfiguration.java
package by.rekuts.travelagency.view;

import by.rekuts.travelagency.domain.User;
import by.rekuts.travelagency.repository.UserSpecification;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.builders.WebSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

@Configuration
@EnableWebSecurity
@EnableGlobalMethodSecurity(prePostEnabled = true)
public class TestSecurityConfiguration extends WebSecurityConfigurerAdapter {
    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http
                .authorizeRequests()
                .antMatchers("/resources/**").permitAll()
                .antMatchers("/", "/sign-in", "/sign-up", "/tours", "/reg-user").permitAll()
                .anyRequest().authenticated()
                .and()
                .exceptionHandling().accessDeniedPage("/error")
                .and()
                .formLogin()
                .loginPage("/sign-in")
                .loginProcessingUrl("/j_spring_security_check")
                .usernameParameter("j_username")
                .passwordParameter("j_password")
                .successHandler((req,res,auth)->{
                    UserSpecification specification = new UserSpecification();
                    specification.setLogin(auth.getName());
                    User curUser = new User();
                    curUser.setId(69);
                    req.getSession().setAttribute("sesUserId", curUser.getId());
                    res.sendRedirect("/");
                })
                .failureHandler((req,res,exp)->{
                    req.setAttribute("message", "Invalid username or password.");
                    res.sendRedirect("/sign-in");
                })
                .permitAll()
                .and()
                .logout()
                .logoutUrl("/sign-out")
                .logoutSuccessHandler((req,res,auth)->{
                    req.setAttribute("message", "You are logged out successfully.");
                    res.sendRedirect("/sign-in");
                })
                .permitAll()
                .and()
                .csrf().disable();
    }

    @Override
    public void configure(WebSecurity web) throws Exception{
        web.debug(true);
    }
}

File: view/src/test/scala/by/rekuts/travelagency/view/configuration/TestSecurityConfiguration.scala
package by.rekuts.travelagency.view.configuration

import by.rekuts.travelagency.domain.User
import by.rekuts.travelagency.repository.UserSpecification
import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.builders.{HttpSecurity, WebSecurity}
import org.springframework.security.config.annotation.web.configuration.{EnableWebSecurity, WebSecurityConfigurerAdapter}
import org.springframework.security.core.{Authentication, AuthenticationException}

@Configuration
@EnableWebSecurity
@EnableGlobalMethodSecurity(prePostEnabled = true)
class TestSecurityConfiguration extends WebSecurityConfigurerAdapter {
  @throws[Exception]
  override protected def configure(http: HttpSecurity): Unit = {
    http
      .authorizeRequests
      .antMatchers("/resources/**").permitAll
      .antMatchers("/", "/sign-in", "/sign-up", "/tours", "/reg-user").permitAll
      .anyRequest.authenticated
      .and
      .exceptionHandling.accessDeniedPage("/error")
      .and
      .formLogin
      .loginPage("/sign-in")
      .loginProcessingUrl("/j_spring_security_check")
      .usernameParameter("j_username")
      .passwordParameter("j_password")
      .successHandler((req: HttpServletRequest, res: HttpServletResponse, auth: Authentication) => {
      def successfulAuthentication(req: HttpServletRequest, res: HttpServletResponse, auth: Authentication) = {
        val specification = new UserSpecification
        specification.login_$eq(auth.getName)
        val curUser = new User
        curUser.id_$eq(69)
        req.getSession.setAttribute("sesUserId", curUser.id)
        res.sendRedirect("/")
      }
      successfulAuthentication(req, res, auth)
    }).failureHandler((req: HttpServletRequest, res: HttpServletResponse, exp: AuthenticationException) => {
      def unsuccessfulAuthentication(req: HttpServletRequest, res: HttpServletResponse, exp: AuthenticationException) = {
        req.setAttribute("message", "Invalid username or password.")
        res.sendRedirect("/sign-in")
      }
      unsuccessfulAuthentication(req, res, exp)
    }).permitAll.and.logout.logoutUrl("/sign-out").logoutSuccessHandler((req: HttpServletRequest, res: HttpServletResponse, auth: Authentication) => {
      def logout(req: HttpServletRequest, res: HttpServletResponse, auth: Authentication) = {
        req.setAttribute("message", "You are logged out successfully.")
        res.sendRedirect("/sign-in")
      }
      logout(req, res, auth)
    }).permitAll.and.csrf.disable
  }

  @throws[Exception]
  override def configure(web: WebSecurity): Unit = {
    web.debug(true)
  }
}
